client/maps/map-preferences-reducer.ts

```diff
@@ -0,0 +1,70 @@
+import { Immutable } from 'immer'
+import { MapPreferences, MapSortType, MapVisibility, NumPlayers, Tileset } from '../../common/maps'
+import { FetchError } from '../network/fetch-action-types'
+import { immerKeyedReducer } from '../reducers/keyed-reducer'
+
+export interface MapPreferencesState {
+  /** The last saved visibility in the maps browser. Can be either official, private, or public.  */
+  visibility: MapVisibility
+  /** The last saved thumbnail size in the maps browser. */
+  thumbnailSize: number
+  /** The last saved map sorting option in the maps browser. */
+  sortOption: MapSortType
+  /** The last saved number of players that was used to filter maps. */
+  numPlayersFilter: Set<NumPlayers>
+  /** The last saved tileset(s) that was used to filter maps. */
+  tilesetFilter: Set<Tileset>
+  /** A flag indicating whether we're currently requesting map preferences. */
+  isRequesting: boolean
+  // NOTE(2Pac): We don't actually display this anywhere since it's not that useful to the user
+  /** A potential error of our request to get map preferences. */
+  lastError?: FetchError
+}
+
+const DEFAULT_STATE: Immutable<MapPreferencesState> = {
+  visibility: MapVisibility.Private,
+  thumbnailSize: 0,
+  sortOption: MapSortType.Name,
+  numPlayersFilter: new Set(),
+  tilesetFilter: new Set(),
+  isRequesting: false,
+  lastError: undefined as FetchError | undefined,
+}
+
+function createPreferences(state: MapPreferencesState, preferences: MapPreferences) {
```

Comment:
This doesn't really create anything, `updatePreferences` is probably a better name